### ISSUE TITLE
zonemaster_backend_testagent log to stdout/stderr

### ISSUE THREAD

**Issue by aabdnn** [zonemaster/zonemaster-backend#612]
Hi folks,

I've just built my custom package of the new 2019.2 release of Zonemaster, to run on CentOS 7. I am starting the test agent using a systemd unit file as follows:

```
[Unit]
Description=Zonemaster Test Agent
After=mariadb.service network-online.target

[Service]
Environment=PERL5LIB=/opt/zonemaster/lib/perl5
User=nobody
ExecStart=/opt/zonemaster/bin/zonemaster_backend_testagent foreground
Restart=always
RestartSec=30

[Install]
WantedBy=multi-user.target
```

The test agent logs to `/var/log/zonemaster/zonemaster_backend_testagent.log` by default. However, since this is running under systemd, I prefer to have it send its logs to stdout or stderr, so that I can view the logs with journalctl.

Now, when I run the testagent from a root shell on the command line, I can trick it into sending logs to stdout, like this:

```
zonemaster_backend_testagent --logfile=/dev/stdout foreground
```

But this doesn't work from within systemd. I get the error:

```
zonemaster_backend_testagent[29265]: Cannot write to '/dev/stdout': No such device or address at /opt/zonemaster/lib/perl5/Log/Dispatch/File.pm line 100.
```

I also tried to specify `/proc/self/fd/1`, but that also doesn't work:

```
zonemaster_backend_testagent[29650]: Cannot write to '/proc/self/fd/1': No such device or address at /opt/zonemaster/lib/perl5/Log/Dispatch/File.pm line 100.
```

How can I make the testagent NOT write to a file, and just output to stdout/stderr instead?

**Comment by matsduf** [comment]
@mtoma, can you take a look at it?

**Comment by aabdnn** [comment]
Hi @mtoma. Do you have any idea for this?

**Comment by aabdnn** [comment]
Hi @matsduf @mtoma any chance of getting this feature in the upcoming 2019.2.1 release?

**Comment by matsduf** [comment]
@aabdnn, no, I'm sorry. I have not forgot it, but I have not got any reply yet.

**Comment by mtoma** [comment]
Hello @aabdnn, the logging to STDOUT and STDERR is currently not supported. The log dispatcher object has a hard coded output of type "File".
If output to STDOUT and STDERR is needed we need to add this as an option to the backend to support the Log::Dispatcher type "Screen".

**Comment by aabdnn** [comment]
Hi @mtoma. I think that logging to stdout and/or syslog would be quite useful. Logging only to a file is rather inflexible. There is also another issue. I want to ensure that the log file is rotated. So I followed the common practice of renaming the log file, and sending the daemon a HUP signal. But that caused the test agent to exit!
I then tried to reload the daemon by calling "zonemaster_backed_testagent reload" but that also sends a HUP signal, and causes the daemon to terminate.
Can you please suggest the right way of rotating the log file without having to restart the daemon?

**Comment by matsduf** [comment]
@aabdnn, thank you for raising the issues. We should definitely try to find a more flexible solution. I add this issue to next release, but it is not a promise, as you understand.

**Comment by aabdnn** [comment]
Thanks @matsduf. I guess in the meantime, I will have to restart the daemon for log file rotation.

**Comment by matsduf** [comment]
@aabdnn, we will discuss this issue further next week and consider it important for next release.

**Comment by matsduf** [comment]
Resolving this issue is prioritized for next release.

**Comment by mattias-p** [comment]
@mtoma I'd like to take over this issue to get it done. Is that okay with you?

**Comment by sandoche2k** [comment]
@mattias-p If you can make a PR then @mtoma will review?

**Comment by matsduf** [comment]
Resolved by #644.
